src/coreclr/src/jit/morph.cpp

```diff
@@ -13354,9 +13354,10 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac)
                 /* Check for "op1 - cns2" , we change it to "op1 + (-cns2)" */
 
                 noway_assert(op2);
-                if (op2->IsCnsIntOrI())
+                if (op2->IsCnsIntOrI() && !op2->IsIConHandle())
```

Comment:
Sorry about that -- here's the right change.
```suggestion
                if (op2->IsCnsIntOrI() && !op2->IsIconHandle())
```